resources/assets/js/features/tripsList/reducer.js

```diff
@@ -0,0 +1,55 @@
+import * as actions from './actionTypes';
+
+const initialState = {
+    trips: [], // list user trips
+    filtered: [], // filtered user trips
+
+    isOpen: false, // status of modal window
+    modalData: [], // data in modal window
+};
+
+
+export default function (state = initialState, action) {
+    switch(action.type) {
+        case actions.TRIPSLIST_GET_ALL: {
+            return {
+                ...state,
+                trips: action.trips
+            };
+        }
+        case actions.TRIPSLIST_FILTER_PAST: {
+            return {
+            ...state,
+            filtered: state.trips.filter(
+                item => (new Date(item.start_at)<new Date()))
```

Comment:
use momentjs for date object manipulation, some browsers could have specific issues with date time parse from string